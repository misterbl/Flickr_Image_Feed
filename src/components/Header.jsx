import React from "react";
import { Formik } from "formik";
import { connect } from "react-redux";
import SearchForm from "../components/SearchForm";
import * as apiThunk from "../actions/thunks/apiThunk";

export class Header extends React.Component {
  searchByTags = data => {
    this.props.getflickrFeedByTags(data.searchByTags);
  };
  render() {
    return (
      <header className="bg-black h3 flex justify-between">
        <p className="white ft-35 b ml6 mt2">flickr</p>
        <div className="ma3 mr6 relative">
          <Formik
            initialValues={{
              searchByTags: ""
            }}
            onSubmit={this.searchByTags}
            render={formikProps => <SearchForm {...formikProps} />}
          />
        </div>
      </header>
    );
  }
}

export const mapDispatchToProps = {
  getflickrFeedByTags: apiThunk.getflickrFeedByTags
};
export default connect(
  null,
  mapDispatchToProps
)(Header);
