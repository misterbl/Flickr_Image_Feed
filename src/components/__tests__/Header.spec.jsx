import * as React from "react";
import { shallow } from "enzyme";
import Header from "../Header";
import historyMock from "../../testMocks/history.mock";
import formikRenderMock from "../../testMocks/formikRender.mock";

describe("Header", () => {
  const props = {
    ...formikRenderMock,
    values: {
      searchByTags: ""
    },
    getflickrFeedByTags: jest.fn(),
    history: historyMock
  };
  const wrapper = shallow(<Header {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
