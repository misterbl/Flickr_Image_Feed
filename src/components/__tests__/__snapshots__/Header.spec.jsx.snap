// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Header)
    dirty={false}
    errors={Object {}}
    getflickrFeedByTags={[MockFunction]}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleReset={[Function]}
    handleSubmit={[Function]}
    isSubmitting={false}
    isValid={false}
    isValidating={false}
    registerField={[Function]}
    resetForm={[Function]}
    setError={[Function]}
    setErrors={[Function]}
    setFieldError={[Function]}
    setFieldTouched={[Function]}
    setFieldValue={[Function]}
    setFormikState={[Function]}
    setStatus={[Function]}
    setSubmitting={[Function]}
    setTouched={[Function]}
    setValues={[Function]}
    submitCount={0}
    submitForm={[Function]}
    touched={Object {}}
    unregisterField={[Function]}
    validateField={[Function]}
    validateForm={[Function]}
    values={
      Object {
        "searchByTags": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
